src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/LæremidlerBeregnYtelseSteg.kt

```diff
@@ -62,32 +69,63 @@ class LæremidlerBeregnYtelseSteg(
         lagreAndeler(saksbehandling, beregningsresultat)
     }
 
-    /*
-    private fun beregnOgLagreOpphør(saksbehandling: Saksbehandling, vedtak: OpphørLæremidlerDto) {
+    private fun beregnOgLagreOpphør(saksbehandling: Saksbehandling, vedtak: OpphørLæremidlerRequest) {
         brukerfeilHvis(saksbehandling.forrigeBehandlingId == null) {
             "Opphør er et ugyldig vedtaksresultat fordi behandlingen er en førstegangsbehandling"
         }
+        val forrigeBehandling = vedtakRepository.findByIdOrNull(saksbehandling.forrigeBehandlingId)
+            ?: error("Opphør må ha en forrige behandling")
+        if (saksbehandling.revurderFra == null) {
+            error("revurderFra-dato er påkrevd for opphør")
+        }
 
-        opphørValideringService.validerPerioder(saksbehandling)
+        val kuttedePerioder: List<BeregningsresultatForMåned> = kuttePerioderVedOpphør(forrigeBehandling, saksbehandling.revurderFra)
 
-        val beregningsresultat = tilsynBarnBeregningService.beregn(saksbehandling, TypeVedtak.OPPHØR)
-        opphørValideringService.validerIngenUtbetalingEtterRevurderFraDato(beregningsresultat, saksbehandling.revurderFra)
         vedtakRepository.insert(
             GeneriskVedtak(
                 behandlingId = saksbehandling.id,
                 type = TypeVedtak.OPPHØR,
                 data = OpphørLæremidler(
-                    beregningsresultat = BeregningsresultatLæremidler(beregningsresultat.perioder),
+                    beregningsresultat = BeregningsresultatLæremidler(kuttedePerioder),
                     årsaker = vedtak.årsakerOpphør,
                     begrunnelse = vedtak.begrunnelse,
                 ),
-
-                ),
+            ),
         )
 
-        lagreAndeler(saksbehandling, beregningsresultat)
+        val beregningsresultatLæremidler = BeregningsresultatLæremidler(kuttedePerioder)
+
+        lagreAndeler(saksbehandling, beregningsresultatLæremidler)
+    }
+
+    private fun kuttePerioderVedOpphør(forrigeBehandling: Vedtak, revurderFra: LocalDate): List<BeregningsresultatForMåned> {
+        val kuttedePerioder: List<BeregningsresultatForMåned> = emptyList()
+
+        when (forrigeBehandling.type) {
```

Comment:
Logikk for å avkorte periodene kunde ha blitt flyttet til `BeregningsresultatLæremidler` for å gjøre det mulig å teste.
Testen skal sjekke:æ
* Tidligere perioder blir med
* Perioder som overlapper med revurderFra blir avkortet
* Fremtidelige perioder blir fjernet

Reply:
cead302f1c8f1ce127dd4ef1d42db0173409086d
2df4ce5c6364854c719c4b4899ce945bd7de6c0b

Reply:
cead302f1c8f1ce127dd4ef1d42db0173409086d
2df4ce5c6364854c719c4b4899ce945bd7de6c0b